src/server/dflycmd.cc

```diff
@@ -353,15 +353,19 @@ void DflyCmd::StopFullSyncInThread(FlowInfo* flow, EngineShard* shard) {
 OpStatus DflyCmd::StartStableSyncInThread(FlowInfo* flow, EngineShard* shard) {
   // Register journal listener and cleanup.
   uint32_t cb_id = 0;
+  JournalWriter* writer = nullptr;
   if (shard != nullptr) {
-    JournalWriter writer{flow->conn->socket()};
-    auto journal_cb = [flow, writer = std::move(writer)](const journal::Entry& je) mutable {
-      writer.Write(je);
+    writer = new JournalWriter{};
```

Comment:
maybe the writer can be in flow struct, as rdb save is in flow

Reply:
There is no need to, it will be replaced anyway in the next pr